lib/livebook/session/data.ex

```diff
@@ -1454,6 +1465,10 @@ defmodule Livebook.Session.Data do
     end
   end
 
+  defp put_secret({data, _} = data_actions, secret) do
+    set!(data_actions, secrets: [secret | data.secrets])
```

Comment:
It's relatively rare, so appending is fine and this way we always preserve the right order

```suggestion
    set!(data_actions, secrets: data.secrets ++ [secret])
```